# **Project Name**    - FedEx Logistics Unpacked: An Exploratory Data Analysis



##### **Project Type**    - EDA
##### **Contribution**    - Individual
##### **Name**            - **Ashwin Suryawanshi**


# **Project Summary -**

This project focuses on analyzing the delivery performance of FedEx logistics operations by examining various factors that influence on-time delivery rates. Using a comprehensive dataset containing shipment details such as shipment mode, country, delivery dates, vendor information, and insurance costs, the analysis uncovers insights into operational efficiencies and bottlenecks.

**Key Objectives:**

	•	Understand whether shipment mode influences on-time delivery.
	•	Identify if certain management teams (like PMO - US) are more effective in ensuring timely deliveries.
	•	Analyze the correlation between shipment weight and insurance costs.
	•	Determine country-specific delivery performance patterns.
	•	Examine the impact of international commercial terms (INCO terms) on vendor delivery reliability.

**Tools & Techniques:**

	•	Python (Pandas, NumPy)
	•	Plotly for interactive visualizations
	•	Exploratory Data Analysis (EDA)
	•	Group-by aggregations and performance metrics
	•	Correlation and distribution analysis

**Major Insights:**

	•	The South Africa Field Office achieved the highest on-time delivery rate (98.25%), while PMO - US lagged behind at 55.67%.
	•	Air shipment mode generally had a better on-time delivery rate compared to Sea and Road.
	•	Significant variation in performance based on country and INCO terms.
	•	A positive correlation observed between shipment weight and insurance costs.


# **GitHub Link -**

https://github.com/engineerash13/Labmentix_Internship_Projects

# **Problem Statement**


To analyze delivery performance, delays, and logistical patterns in the FedEx supply chain using historical shipment data, and identify key factors affecting on-time delivery.

#### **Define Your Business Objective?**

FedEx Logistics operates a vast and complex global supply chain, handling shipments across multiple regions, industries, and transportation modes. The provided dataset captures critical aspects of this network, including purchase orders, delivery schedules, shipment modes, INCO terms, and product-level details such as item descriptions and dosage forms.

In an era dominated by eCommerce and global distribution, FedEx faces increasing pressure to deliver efficiently, minimize freight costs, and ensure customer satisfaction. Analyzing this dataset provides an opportunity to:

* Understand how shipments are managed.

* Identify common causes of delays or inefficiencies.

* Optimize delivery timelines.

* Reduce logistical costs.

The goal is to support data-driven decisions that enhance the overall performance and reliability of FedEx’s supply chain operations.



# **General Guidelines** : -  

1.   Well-structured, formatted, and commented code is required.
2.   Exception Handling, Production Grade Code & Deployment Ready Code will be a plus. Those students will be awarded some additional credits.
     
     The additional credits will have advantages over other students during Star Student selection.
       
             [ Note: - Deployment Ready Code is defined as, the whole .ipynb notebook should be executable in one go
                       without a single error logged. ]

3.   Each and every logic should have proper comments.
4. You may add as many number of charts you want. Make Sure for each and every chart the following format should be answered.
        

```
# Chart visualization code
```
            

*   Why did you pick the specific chart?
*   What is/are the insight(s) found from the chart?
* Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

5. You have to create at least 20 logical & meaningful charts having important insights.


[ Hints : - Do the Vizualization in  a structured way while following "UBM" Rule.

U - Univariate Analysis,

B - Bivariate Analysis (Numerical - Categorical, Numerical - Numerical, Categorical - Categorical)

M - Multivariate Analysis
 ]





# ***Let's Begin !***

## ***1. Know Your Data***

### Import Libraries

In [ ]:
# Import Libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [ ]:
# For handling warnings
import warnings
warnings.filterwarnings('ignore')

# For displaying all columns
pd.set_option('display.max_columns', None)

### Dataset Loading

In [ ]:
# Load Dataset
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df=pd.read_csv("/content/drive/MyDrive/SCMS_Delivery_History_Dataset.csv")

### Dataset First View

In [ ]:
# Dataset First Look
df.head()

,ID,Project Code,PQ #,PO / SO #,ASN/DN #,Country,Managed By,Fulfill Via,Vendor INCO Term,Shipment Mode,PQ First Sent to Client Date,PO Sent to Vendor Date,Scheduled Delivery Date,Delivered to Client Date,Delivery Recorded Date,Product Group,Sub Classification,Vendor,Item Description,Molecule/Test Type,Brand,Dosage,Dosage Form,Unit of Measure (Per Pack),Line Item Quantity,Line Item Value,Pack Price,Unit Price,Manufacturing Site,First Line Designation,Weight (Kilograms),Freight Cost (USD),Line Item Insurance (USD)
0,1,100-CI-T01,Pre-PQ Process,SCMS-4,ASN-8,Côte d'Ivoire,PMO - US,Direct Drop,EXW,Air,Pre-PQ Process,Date Not Captured,2-Jun-06,2-Jun-06,2-Jun-06,HRDT,HIV test,RANBAXY Fine Chemicals LTD.,"HIV, Reveal G3 Rapid HIV-1 Antibody Test, 30 T...","HIV, Reveal G3 Rapid HIV-1 Antibody Test",Reveal,NaN,Test kit,30,19,551.0,29.00,0.97,Ranbaxy Fine Chemicals LTD,Yes,13,780.34,NaN
1,3,108-VN-T01,Pre-PQ Process,SCMS-13,ASN-85,Vietnam,PMO - US,Direct Drop,EXW,Air,Pre-PQ Process,Date Not Captured,14-Nov-06,14-Nov-06,14-Nov-06,ARV,Pediatric,Aurobindo Pharma Limited,"Nevirapine 10mg/ml, oral suspension, Bottle, 2...",Nevirapine,Generic,10mg/ml,Oral suspension,240,1000,6200.0,6.20,0.03,"Aurobindo Unit III, India",Yes,358,4521.5,NaN
2,4,100-CI-T01,Pre-PQ Process,SCMS-20,ASN-14,Côte d'Ivoire,PMO - US,Direct Drop,FCA,Air,Pre-PQ Process,Date Not Captured,27-Aug-06,27-Aug-06,27-Aug-06,HRDT,HIV test,Abbott GmbH & Co. KG,"HIV 1/2, Determine Complete HIV Kit, 100 Tests","HIV 1/2, Determine Complete HIV Kit",Determine,NaN,Test kit,100,500,40000.0,80.00,0.80,ABBVIE GmbH & Co.KG Wiesbaden,Yes,171,1653.78,NaN
3,15,108-VN-T01,Pre-PQ Process,SCMS-78,ASN-50,Vietnam,PMO - US,Direct Drop,EXW,Air,Pre-PQ Process,Date Not Captured,1-Sep-06,1-Sep-06,1-Sep-06,ARV,Adult,SUN PHARMACEUTICAL INDUSTRIES LTD (RANBAXY LAB...,"Lamivudine 150mg, tablets, 60 Tabs",Lamivudine,Generic,150mg,Tablet,60,31920,127360.8,3.99,0.07,"Ranbaxy, Paonta Shahib, India",Yes,1855,16007.06,NaN
4,16,108-VN-T01,Pre-PQ Process,SCMS-81,ASN-55,Vietnam,PMO - US,Direct Drop,EXW,Air,Pre-PQ Process,Date Not Captured,11-Aug-06,11-Aug-06,11-Aug-06,ARV,Adult,Aurobindo Pharma Limited,"Stavudine 30mg, capsules, 60 Caps",Stavudine,Generic,30mg,Capsule,60,38000,121600.0,3.20,0.05,"Aurobindo Unit III, India",Yes,7590,45450.08,NaN


### Dataset Rows & Columns count

In [ ]:
# Dataset Rows & Columns count

print(f"Number of Rows: {df.shape[0]}")
print(f"Number of Columns: {df.shape[1]}")

Number of Rows: 10324
Number of Columns: 33


### Dataset Information

In [ ]:
# Dataset Info
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10324 entries, 0 to 10323
Data columns (total 33 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   ID                            10324 non-null  int64  
 1   Project Code                  10324 non-null  object 
 2   PQ #                          10324 non-null  object 
 3   PO / SO #                     10324 non-null  object 
 4   ASN/DN #                      10324 non-null  object 
 5   Country                       10324 non-null  object 
 6   Managed By                    10324 non-null  object 
 7   Fulfill Via                   10324 non-null  object 
 8   Vendor INCO Term              10324 non-null  object 
 9   Shipment Mode                 9964 non-null   object 
 10  PQ First Sent to Client Date  10324 non-null  object 
 11  PO Sent to Vendor Date        10324 non-null  object 
 12  Scheduled Delivery Date       10324 non-null  object 
 13  D

#### Duplicate Values

In [ ]:
# Dataset Duplicate Value Count
print(df.duplicated().sum())

0


#### Missing Values/Null Values

In [ ]:
# Missing Values/Null Values Count
df.isnull().sum()

,0
ID,0
Project Code,0
PQ #,0
PO / SO #,0
ASN/DN #,0
Country,0
Managed By,0
Fulfill Via,0
Vendor INCO Term,0
Shipment Mode,360


In [ ]:
missing_summary = pd.DataFrame({
    'Missing Values': df.isnull().sum(),
    'Missing Percentage': (df.isnull().sum() / len(df)) * 100
})

# Include column names
missing_summary.index.name = 'Column Name'

# Reset index to make it look like a proper table
missing_summary = missing_summary.reset_index()

# Sort by missing values in descending order
missing_summary = missing_summary.sort_values(by='Missing Values', ascending=False)

print(missing_summary)

                     Column Name  Missing Values  Missing Percentage
21                        Dosage            1736           16.815188
9                  Shipment Mode             360            3.487021
32     Line Item Insurance (USD)             287            2.779930
0                             ID               0            0.000000
1                   Project Code               0            0.000000
5                        Country               0            0.000000
6                     Managed By               0            0.000000
3                      PO / SO #               0            0.000000
2                           PQ #               0            0.000000
8               Vendor INCO Term               0            0.000000
10  PQ First Sent to Client Date               0            0.000000
11        PO Sent to Vendor Date               0            0.000000
12       Scheduled Delivery Date               0            0.000000
13      Delivered to Client Date  

In [ ]:
# Visualizing the missing values

missing_data = df.isnull()   #Assuming missing data

fig = px.imshow(missing_data,
                color_continuous_scale='PuBuGn',  # Choose color scale
                title="Missing Data Heatmap")

# Update layout for better clarity
fig.update_layout(
    xaxis_title="Columns",
    yaxis_title="Rows",
    xaxis={'showticklabels': True},
    yaxis={'showticklabels': True},
)

fig.show()

### What did you know about your dataset?

* The dataset contains 10,324 rows and 33 columns.

* There are 3 columns with int64 data type, 4 columns with float64, and 26 columns with object type.

* No duplicate rows are present in the dataset.

* The Dosage column has 1,736 missing values, which is about 16.82% of the data.

* The Shipment Mode column has 360 missing values (3.49%).

* The Line Item Insurance (USD) column has 287 missing values (2.78%).

* Key identifier fields include ID, Project Code, PQ #, and PO / SO #.

* Date-related columns include PO Sent to Vendor Date, Scheduled Delivery Date, and Delivered to Client Date.

* Important numeric columns for analysis include Unit Price, Line Item Value, Freight Cost (USD), and Pack Price.

* Product-related details are found in columns like Item Description, Brand, Dosage Form, and Molecule/Test Type.

## ***2. Understanding Your Variables***

In [ ]:
# Dataset Columns
df.columns

Index(['ID', 'Project Code', 'PQ #', 'PO / SO #', 'ASN/DN #', 'Country',
       'Managed By', 'Fulfill Via', 'Vendor INCO Term', 'Shipment Mode',
       'PQ First Sent to Client Date', 'PO Sent to Vendor Date',
       'Scheduled Delivery Date', 'Delivered to Client Date',
       'Delivery Recorded Date', 'Product Group', 'Sub Classification',
       'Vendor', 'Item Description', 'Molecule/Test Type', 'Brand', 'Dosage',
       'Dosage Form', 'Unit of Measure (Per Pack)', 'Line Item Quantity',
       'Line Item Value', 'Pack Price', 'Unit Price', 'Manufacturing Site',
       'First Line Designation', 'Weight (Kilograms)', 'Freight Cost (USD)',
       'Line Item Insurance (USD)'],
      dtype='object', name='Column Name')

In [ ]:
# Dataset Describe
df.describe()

Column Name,ID,Unit of Measure (Per Pack),Line Item Quantity,Line Item Value,Pack Price,Unit Price,Line Item Insurance (USD)
count,10324.000000,10324.000000,10324.000000,1.032400e+04,10324.000000,10324.000000,10037.000000
mean,51098.968229,77.990895,18332.534870,1.576506e+05,21.910241,0.611701,240.117626
std,31944.332496,76.579764,40035.302961,3.452921e+05,45.609223,3.275808,500.190568
min,1.000000,1.000000,1.000000,0.000000e+00,0.000000,0.000000,0.000000
25%,12795.750000,30.000000,408.000000,4.314593e+03,4.120000,0.080000,6.510000
50%,57540.500000,60.000000,3000.000000,3.047147e+04,9.300000,0.160000,47.040000
75%,83648.250000,90.000000,17039.750000,1.664471e+05,23.592500,0.470000,252.400000
max,86823.000000,1000.000000,619999.000000,5.951990e+06,1345.640000,238.650000,7708.440000


### Variables Description

1. ID -
Unique identifier for each logistics record.

2. Project Code -
Code representing the project or initiative tied to the shipment.

3. PQ # -
Price Quotation reference number.

4. PO / SO # -
Purchase Order or Sales Order associated with the shipment.

5. ASN/DN # -
Advanced Shipment Notice or Delivery Note reference.

6. Country -
Destination country for the shipment.

7. Managed By -
Team responsible for managing the project.

8. Fulfill Via -
Method of fulfillment.

9. Vendor INCO Term -
Incoterms defining the shipping agreement).

10. Shipment Mode -
Mode of transportation used.

11. PQ First Sent to Client Date -
Date the Price Quotation was first sent to the client.

12. PO Sent to Vendor Date -
Date the Purchase Order was sent to the vendor.

13. Scheduled Delivery Date -
Planned date for the delivery.

14. Delivered to Client Date -
Actual date the delivery was made to the client.

15. Delivery Recorded Date -
Date the delivery was logged/recorded in the system.

16. Product Group -
High-level category of the product.

17. Sub Classification -
More specific classification within the product group.

18. Vendor -
Name of the supplier or manufacturer.

19. Item Description -
Description of the item being shipped.

20. Molecule/Test Type -
Type of molecule or diagnostic test involved.

21. Brand -
Brand name of the product.

22. Dosage -
Dosage strength of the pharmaceutical product.

23. Dosage Form -
Physical form in which the product is delivered.

24. Unit of Measure (Per Pack) -
Number of units contained in one pack.

25. Line Item Quantity -
Total quantity of units being shipped.

26. Line Item Value -
Total value of the shipment line item.

27. Pack Price -
Price per pack.

28. Unit Price -
Price per individual unit.

29. Manufacturing Site -
Location or facility where the item was produced.

30. First Line Designation -
Whether the product is intended for first-line use in treatment.

31. Weight (Kilograms) -
Weight of the shipment in kilograms.

32. Freight Cost (USD) -
Cost of shipping in U.S. dollars.

33. Line Item Insurance (USD) -
Insurance cost for the line item.

### Check Unique Values for each variable.

In [ ]:
# Check Unique Values for each variable.
df.nunique().sort_values(ascending=False)

,0
Column Name,
ID,10324
Line Item Value,8741
ASN/DN #,7030
Freight Cost (USD),6733
Line Item Insurance (USD),6722
PO / SO #,6233
Line Item Quantity,5065
Weight (Kilograms),4688
Delivered to Client Date,2093


## 3. ***Data Wrangling***

### Data Wrangling Code

In [ ]:
# Write your code to make your dataset analysis ready.

df1=df.copy()

In [ ]:
# 1. Handle Missing Values

# Drop rows with missing 'Dosage' since it has the highest % of missing values (16.8%)
df1 = df1.dropna(subset=['Dosage'])

# Fill missing 'Shipment Mode' with the mode (most frequent value)
df1['Shipment Mode'].fillna(df1['Shipment Mode'].mode()[0], inplace=True)

# Fill missing 'Line Item Insurance (USD)' with 0 or mean/median
df1['Line Item Insurance (USD)'].fillna(0, inplace=True)

In [ ]:
# 2. Convert Columns to Correct Data Types

# Remove commas/dollar signs if present and convert to float
df1['Weight (Kilograms)'] = pd.to_numeric(df1['Weight (Kilograms)'].str.replace(',', ''), errors='coerce')
df1['Freight Cost (USD)'] = pd.to_numeric(df1['Freight Cost (USD)'].str.replace(',', ''), errors='coerce')


In [ ]:
# 3. Convert Date Columns to datetime

date_cols = [
    'Delivery Recorded Date',
    'Delivered to Client Date',
    'Scheduled Delivery Date',
    'PO Sent to Vendor Date',
    'PQ First Sent to Client Date'
]

for col in date_cols:
    df1[col] = pd.to_datetime(df1[col], errors='coerce')


In [ ]:
# 4. Create New Useful Columns

# Example: Calculate delivery delay
df1['Delivery Delay (Days)'] = (df1['Delivered to Client Date'] - df1['Scheduled Delivery Date']).dt.days


In [ ]:
 # 5. Drop Duplicates

 df1.drop_duplicates(inplace=True)


In [ ]:
# 6. Reset Index (after dropping rows)

df1.reset_index(drop=True, inplace=True)

In [ ]:
df1.head()

Column Name,ID,Project Code,PQ #,PO / SO #,ASN/DN #,Country,Managed By,Fulfill Via,Vendor INCO Term,Shipment Mode,PQ First Sent to Client Date,PO Sent to Vendor Date,Scheduled Delivery Date,Delivered to Client Date,Delivery Recorded Date,Product Group,Sub Classification,Vendor,Item Description,Molecule/Test Type,Brand,Dosage,Dosage Form,Unit of Measure (Per Pack),Line Item Quantity,Line Item Value,Pack Price,Unit Price,Manufacturing Site,First Line Designation,Weight (Kilograms),Freight Cost (USD),Line Item Insurance (USD),Delivery Delay (Days)
0,3,108-VN-T01,Pre-PQ Process,SCMS-13,ASN-85,Vietnam,PMO - US,Direct Drop,EXW,Air,NaT,NaT,2006-11-14,2006-11-14,2006-11-14,ARV,Pediatric,Aurobindo Pharma Limited,"Nevirapine 10mg/ml, oral suspension, Bottle, 2...",Nevirapine,Generic,10mg/ml,Oral suspension,240,1000,6200.0,6.20,0.03,"Aurobindo Unit III, India",Yes,358.0,4521.50,0.0,0
1,15,108-VN-T01,Pre-PQ Process,SCMS-78,ASN-50,Vietnam,PMO - US,Direct Drop,EXW,Air,NaT,NaT,2006-09-01,2006-09-01,2006-09-01,ARV,Adult,SUN PHARMACEUTICAL INDUSTRIES LTD (RANBAXY LAB...,"Lamivudine 150mg, tablets, 60 Tabs",Lamivudine,Generic,150mg,Tablet,60,31920,127360.8,3.99,0.07,"Ranbaxy, Paonta Shahib, India",Yes,1855.0,16007.06,0.0,0
2,16,108-VN-T01,Pre-PQ Process,SCMS-81,ASN-55,Vietnam,PMO - US,Direct Drop,EXW,Air,NaT,NaT,2006-08-11,2006-08-11,2006-08-11,ARV,Adult,Aurobindo Pharma Limited,"Stavudine 30mg, capsules, 60 Caps",Stavudine,Generic,30mg,Capsule,60,38000,121600.0,3.20,0.05,"Aurobindo Unit III, India",Yes,7590.0,45450.08,0.0,0
3,23,112-NG-T01,Pre-PQ Process,SCMS-87,ASN-57,Nigeria,PMO - US,Direct Drop,EXW,Air,NaT,NaT,2006-09-28,2006-09-28,2006-09-28,ARV,Pediatric,Aurobindo Pharma Limited,"Zidovudine 10mg/ml, oral solution, Bottle, 240 ml",Zidovudine,Generic,10mg/ml,Oral solution,240,416,2225.6,5.35,0.02,"Aurobindo Unit III, India",Yes,504.0,5920.42,0.0,0
4,44,110-ZM-T01,Pre-PQ Process,SCMS-139,ASN-130,Zambia,PMO - US,Direct Drop,DDU,Air,NaT,NaT,2007-01-08,2007-01-08,2007-01-08,ARV,Pediatric,MERCK SHARP & DOHME IDEA GMBH (FORMALLY MERCK ...,"Efavirenz 200mg [Stocrin/Sustiva], capsule, 90...",Efavirenz,Stocrin/Sustiva,200mg,Capsule,90,135,4374.0,32.40,0.36,MSD South Granville Australia,Yes,328.0,NaN,0.0,0


### What all manipulations have you done and insights you found?


#**Data Wrangling Steps**

**1. Handled Missing Values**

* Dropped rows with missing Dosage (~16.8% missing).

* Filled missing values in Shipment Mode using mode (most frequent category).

* Replaced missing Line Item Insurance (USD) with 0.

**2. Converted Data Types**

* Converted Weight (Kilograms) and Freight Cost (USD) from object to numeric after removing commas.

* Parsed date columns to datetime format:

 * Delivery Recorded Date

 * Delivered to Client Date

 * Scheduled Delivery Date

 * PO Sent to Vendor Date

 * PQ First Sent to Client Date

**3. Created New Columns**

* Calculated Delivery Delay (Days) by subtracting Scheduled Delivery Date from Delivered to Client Date.

**4. Removed Duplicates**

* Checked and dropped any duplicate rows.

**5. Reset Index**

* Reindexed the DataFrame after row drops for clean structure.

#**Initial Insights**

**1. Data Completeness**

* Dosage, Shipment Mode, and Line Item Insurance (USD) had missing values.

* Most other columns were complete and clean.

**2. Inconsistent Data Types**

* Weight and freight cost were stored as text, possibly affecting analysis.

* Fixed by converting to numeric format.

**3. Potential for Delivery Analysis**

* With Delivery Delay (Days), we can now analyze logistics efficiency across countries, vendors, or shipment modes.

**4. Structured Temporal Data**

* Cleaned date columns now enable timeline-based analysis (e.g., order-to-delivery durations, seasonal trends).



## ***4. Data Vizualization, Storytelling & Experimenting with charts : Understand the relationships between variables***

#### Chart - 1 Top 15 Countries by Number of Deliveries (Bar Chart)

In [ ]:
# Chart - 1 visualization code

# Get top 15 countries by number of deliveries
top_countries = df1['Country'].value_counts().nlargest(15).reset_index()
top_countries.columns = ['Country', 'Number of Deliveries']

# Create bar chart
fig = px.bar(top_countries,
             x='Country',
             y='Number of Deliveries',
             title='Top 15 Countries by Number of Deliveries',
             color='Number of Deliveries',
             text='Number of Deliveries')

fig.update_traces(textposition='outside')
fig.update_layout(xaxis_title='Country', yaxis_title='Number of Deliveries')

fig.show()

##### 1. Why did you pick the specific chart?

To compare delivery volumes across countries. Bar charts are ideal for showing category-wise counts.

##### 2. What is/are the insight(s) found from the chart?

Identifies which countries received the most deliveries. Shows where logistics activity is highest (e.g. South Africa, Nigeria).

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Business Impact:**

Helps optimize supply chain, resource allocation, and demand forecasting in high-volume regions.

**Potential Negative Insight:**

Over-reliance on a few countries can create risk. Uneven distribution may signal imbalance in market coverage.

#### Chart - 2 Distribution of Shipment Modes (Pie Chart)

In [ ]:
# Chart - 2 visualization code

# Prepare data
shipment_mode_counts = df1['Shipment Mode'].value_counts().reset_index()
shipment_mode_counts.columns = ['Shipment Mode', 'Count']

# Plot pie chart
fig = px.pie(shipment_mode_counts,
             names='Shipment Mode',
             values='Count',
             title='Distribution of Shipment Modes',
             color_discrete_sequence=px.colors.qualitative.Pastel)

fig.show()


##### 1. Why did you pick the specific chart?

A pie chart is ideal to show proportion or share of categories in a whole. It helps visualize how different shipment modes contribute to total deliveries.

##### 2. What is/are the insight(s) found from the chart?

* Air is the most commonly used shipment mode with 55.8% followed by Truck with 32.4%

* Shows whether shipment methods are diverse or concentrated.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Business Impact:**

* Helps optimize logistics cost by identifying overused or underused shipment methods.

* Aids in planning alternative modes for flexibility and cost efficiency.

**Potential Negative Insight:**

* If one mode (e.g., Air) dominates, it may indicate higher transport costs and less resilience in the supply chain.

* Heavy reliance on a single mode can cause delays during disruptions (e.g., airport closures, shipping backlogs).

#### Chart - 3 Distribution of Delivery Delays (Histogram)

In [ ]:
# Chart - 3 visualization code

fig = px.histogram(df1,
                   x='Delivery Delay (Days)',
                   nbins=30,
                   title='Distribution of Delivery Delays (in Days)',
                   color_discrete_sequence=['#636EFA'])

fig.update_layout(xaxis_title='Delivery Delay (Days)',
                  yaxis_title='Number of Deliveries')

fig.show()


##### 1. Why did you pick the specific chart?

A histogram is perfect for analyzing distribution of numerical data—in this case, the number of days a delivery was early or late.

##### 2. What is/are the insight(s) found from the chart?

* Most deliveries occurred close to the scheduled date (around 0 days delay).

* A few deliveries had extreme early or late delays (e.g., beyond ±100 days), possibly indicating data entry errors or exceptional cases.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Business Impact:**

* Majority of deliveries are on time or slightly delayed, indicating efficient logistics.

* Helps identify acceptable delay ranges and plan for buffer periods.

**Potential Negative Insight:**

* The presence of extreme delays (especially on the negative side) suggests possible data quality issues or mismanagement in certain cases.

* Such anomalies need investigation as they could impact reporting accuracy and trust.



#### Chart - 4 Delivery Delay Distribution by Shipment Mode (Box Plot)

In [ ]:
# Chart - 4 visualization code

# Plot box plot
fig = px.box(df1,
             x='Shipment Mode',
             y='Delivery Delay (Days)',
             title='Delivery Delay Distribution by Shipment Mode',
             color='Shipment Mode',
             color_discrete_sequence=px.colors.qualitative.Set2)

fig.update_layout(xaxis_title='Shipment Mode',
                  yaxis_title='Delivery Delay (Days)')

fig.show()

##### 1. Why did you pick the specific chart?

A box plot is ideal for comparing distribution, spread, and outliers across multiple categories. Here, it clearly shows how delivery delays vary between different shipment modes like Air, Truck, Ocean, and Air Charter.

##### 2. What is/are the insight(s) found from the chart?

* Air Charter shows the most consistent delivery times with minimal variation.

* Truck and Air modes have wider delay ranges and more outliers, indicating occasional extreme delays.

* Ocean shipments, while less volatile, still show a moderate number of delays.

* Negative values indicate early deliveries, while positive values reflect delays.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Impact:**

* Identifying Air Charter as the most reliable mode allows for better planning in time-critical deliveries.

* Helps the supply chain team optimize shipment strategies based on urgency.

**Negative Insight:**

* Truck and Air modes show significant variability and outliers (e.g., >100 days), which may lead to unpredictable delivery timelines.

* These delays may affect inventory management, client satisfaction, and costs due to expedited replacements or penalties.

#### Chart - 5 Top 10 Vendors by Average Delivery Delay (Bar Chart)

In [ ]:
# Chart - 5 visualization code

# Group by Vendor and calculate average delivery delay
vendor_delay = df1.groupby('Vendor')['Delivery Delay (Days)'].mean().reset_index()

# Sort and filter top 10 vendors with highest average delays
top_vendor_delays = vendor_delay.sort_values(by='Delivery Delay (Days)', ascending=False).head(10)

# Plot
fig5 = px.bar(
    top_vendor_delays,
    x='Delivery Delay (Days)',
    y='Vendor',
    orientation='h',
    title='Top 10 Vendors by Average Delivery Delay (Days)',
    color='Delivery Delay (Days)',
    color_continuous_scale='Reds'
)

fig5.update_layout(
    xaxis_title='Average Delay (Days)',
    yaxis_title='Vendor Name',
    yaxis={'categoryorder': 'total ascending'}
)

fig5.show()


##### 1. Why did you pick the specific chart?

I chose this horizontal bar chart because it effectively compares the average delivery delays across multiple vendors in a clear, organized, and visually accessible manner. The horizontal layout ensures that vendor names, which can be lengthy, are easily readable, and the color gradient helps quickly identify vendors with higher delays.

##### 2. What is/are the insight(s) found from the chart?

* CIPLA LIMITED and Aurobindo Pharma Limited have the highest average delivery delays (around 5 days).
* Vendors like EMCURE PHARMACEUTICALS LTD and STRIDES ARCOLAB LIMITED also show notable delays.
* Vendors like ABBOTT LOGISTICS B.V. and ABBOTT LABORATORIES (PUERTO RICO) have minimal delivery delays.
* There’s a noticeable gap in performance among vendors, indicating inconsistency in delivery timelines.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Business Impact:**

* Helps identify vendors causing the most delays.
* Enables management to initiate corrective actions (negotiations, penalties, or vendor replacements).
* Improves operational efficiency and customer satisfaction by ensuring timely deliveries.
* Supports better decision-making in vendor selection and management.

**Insights Leading to Negative Growth:**

* Continuous high delivery delays from major vendors like CIPLA LIMITED and Aurobindo Pharma Limited can cause:
* Supply chain disruptions.
* Increased operational costs.
* Poor customer experience.
* Long-term business loss if not addressed.

#### Chart - 6 On-Time Delivery Percentage by Fulfill Via Mode (Bar Chart)

In [ ]:
# Chart - 6 visualization code

In [ ]:

# Create an on-time status column
df1['On Time'] = df1['Delivery Delay (Days)'] <= 0

# Group by Fulfill Via and calculate percentage of on-time deliveries
fulfill_ontime = df1.groupby('Fulfill Via')['On Time'].mean().reset_index()
fulfill_ontime['On Time %'] = fulfill_ontime['On Time'] * 100

# Plot
fig6 = px.bar(
    fulfill_ontime,
    x='Fulfill Via',
    y='On Time %',
    title='On-Time Delivery Percentage by Fulfill Via Mode',
    text='On Time %',
    color='On Time %',
    color_continuous_scale='Blues'
)

fig6.update_layout(
    yaxis_title='On-Time Delivery (%)',
    xaxis_title='Fulfill Via Method',
    uniformtext_minsize=8,
    uniformtext_mode='hide'
)

fig6.show()


##### 1. Why did you pick the specific chart?

I chose a bar chart because it clearly compares on-time delivery percentages between two fulfillment methods:

* Direct Drop

* From RDC

Bar charts are highly effective for visualizing percentage-based comparisons, making it easy to spot performance differences across categories. In this case, the bar chart quickly shows which fulfillment method is more reliable in delivering on time.

##### 2. What is/are the insight(s) found from the chart?

* Direct Drop fulfillment achieves a higher on-time delivery rate (95.66%) compared to From RDC (82.26%).

* The 13.4% gap suggests that Direct Drop is significantly more efficient and reliable in meeting delivery deadlines.

* From RDC has a comparatively lower on-time performance, which could signal operational inefficiencies or logistics delays.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Impact:**

* The company can prioritize Direct Drop for high-priority or time-sensitive shipments, improving customer satisfaction.

* Insights allow operations teams to benchmark performance and drive efficiency in underperforming fulfillment methods.

* Helps allocate resources to the most reliable logistics strategies, improving delivery KPIs and brand trust.

**Negative Impact:**

* Continuing to rely heavily on From RDC without improvements could lead to increased late deliveries, customer complaints, and brand damage.

* Without action, the company might miss opportunities to streamline operations or reduce costs associated with delays and re-shipping.

#### Chart - 7 Monthly Delivery Trend - On Time vs Late (Line Chart)

In [ ]:
# Convert date columns to datetime
df1['Scheduled Delivery Date'] = pd.to_datetime(df1['Scheduled Delivery Date'], errors='coerce')
df1['Delivered to Client Date'] = pd.to_datetime(df1['Delivered to Client Date'], errors='coerce')

# Derive Delivery Status
df1['Delivery Status'] = df1['Delivered to Client Date'] <= df1['Scheduled Delivery Date']
df1['Month'] = df1['Delivered to Client Date'].dt.to_period('M').astype(str)

# Count deliveries by month and status
monthly_trend = df1.groupby(['Month', 'Delivery Status']).size().reset_index(name='Count')

# Plotting
fig = px.line(
    monthly_trend,
    x='Month',
    y='Count',
    color='Delivery Status',
    markers=True,
    title='Monthly Delivery Trend - On Time vs Late',
    color_discrete_map={True: 'green', False: 'red'}
)

fig.update_layout(
    xaxis_title='Month',
    yaxis_title='Number of Deliveries',
    legend_title='On Time'
)

fig.show()



##### 1. Why did you pick the specific chart?

I chose a line chart to visualize the trend of delivery performance over time, specifically distinguishing between on-time and late deliveries on a monthly basis. Line charts are ideal for time-series data, allowing us to observe fluctuations, patterns, and seasonality in delivery behavior across years.

##### 2. What is/are the insight(s) found from the chart?

* On-time deliveries have consistently been higher than late deliveries throughout the years.

* However, there is significant fluctuation in both metrics over time, especially in the green line, indicating volatility in performance.

* There are occasional spikes in late deliveries, especially in certain months (e.g., post-2010), which could point to supply chain disruptions or external challenges.

* From 2014 onward, a general increase in late deliveries is noticeable, which could indicate worsening performance or increased demand that overwhelmed logistics capacity.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Business Impact:**

* Trend analysis enables the company to identify months or years with exceptional performance, which can be studied and replicated.

* The visual highlights problem periods, prompting deeper investigation into logistics bottlenecks, vendor issues, or seasonal challenges.

* Enables data-driven decision-making for process improvement, resource allocation, and vendor management.

**Negative Impact:**

* Ignoring this trend could lead to customer dissatisfaction due to persistent delivery delays.

* Long-term reputation damage may occur if the late delivery trend increases.

* Poor logistics planning without this insight can result in higher operational costs and inventory mismanagement.

#### Chart - 8 Top 10 Countries by Average Delivery Time (Bar Chart)

In [ ]:
# Chart - 8 visualization code

#Calculate the Delivery Time (Days) as the difference between the recorded and scheduled dates
df1['Delivery Time (Days)'] = (df1['Delivery Recorded Date'] - df1['Scheduled Delivery Date']).dt.days

# Group the data by 'Country' and compute the average delivery time, then sort in descending order
avg_delivery_by_country = df1.groupby('Country')['Delivery Time (Days)'].mean().sort_values(ascending=False).head(10)

# Create a horizontal bar chart using Plotly Express
fig = px.bar(
    avg_delivery_by_country,
    x=avg_delivery_by_country.values,
    y=avg_delivery_by_country.index,
    orientation='h',
    labels={'x': 'Average Delivery Time (Days)', 'y': 'Country'},
    title='Top 10 Countries by Average Delivery Time'
)

fig.update_layout(
    xaxis_title='Average Delivery Time (Days)',
    yaxis_title='Country'
)

fig.show()

##### 1. Why did you pick the specific chart?

This horizontal bar chart was chosen because it provides a clear and easy-to-read comparison of average delivery times across countries. Bar charts are ideal for ranking and comparing numerical values across categories — in this case, average delivery time (in days) for the top 10 countries. The horizontal layout improves label visibility and highlights variations more effectively, especially for country names that may be long.

##### 2. What is/are the insight(s) found from the chart?

* Congo, DRC, Togo, and Burundi have the highest average delivery times, exceeding 18 to 24 days, which is significantly longer than other countries.

* In contrast, countries like Senegal and Botswana have shorter average delivery durations, indicating relatively better logistics performance.

* There is a wide disparity in delivery performance across countries, suggesting inconsistent supply chain effectiveness in different regions.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Business Impact:**

* The insights can help prioritize improvement efforts in logistics and vendor selection in underperforming countries.

* By focusing on countries with long delivery times, the company can optimize supply chain strategies, improve vendor reliability, and enhance customer satisfaction.

* Can be used to inform resource allocation, such as more efficient shipping routes or local warehousing.

**Negative Impact / Risk of Negative Growth:**

* High delivery times in key countries (like Congo, DRC and Togo) could indicate systemic inefficiencies, such as customs delays, unreliable vendors, or poor infrastructure.

* If not addressed, these delays may lead to customer dissatisfaction, increased operational costs, and loss of contracts or partnerships in those regions.

* Delays could also disrupt healthcare or essential supply chains, especially if the deliveries involve medical supplies.

#### Chart - 9 Average Delivery Delay by Vendor and Product Group (Density Heatmap)

In [ ]:
# Chart - 9 visualization code

# Group by Vendor and Product Group and calculate the mean delay
heatmap_data = df1.groupby(['Vendor', 'Product Group'])['Delivery Delay (Days)'].mean().reset_index()

# Create a heatmap using Plotly
fig = px.density_heatmap(
    heatmap_data,
    x='Product Group',
    y='Vendor',
    z='Delivery Delay (Days)',
    color_continuous_scale='Reds',
    title='Average Delivery Delay by Vendor and Product Group'
)

# Update axis labels
fig.update_layout(
    xaxis_title='Product Group',
    yaxis_title='Vendor'
)

# Show the heatmap
fig.show()

##### 1. Why did you pick the specific chart?

A heatmap is ideal for visualizing patterns and comparing two categorical variables (Vendor & Product Group) against a quantitative variable (Average Delivery Delay).
This chart makes it very easy to spot vendors and product categories where delays are higher or lower, using intuitive color intensity.

##### 2. What is/are the insight(s) found from the chart?

* Certain vendors consistently deliver earlier (negative delays), especially in specific product groups like ARV or ANTM.

* A few vendors show positive delay values, signaling late deliveries (though rare).

* The delays are not uniform across product groups, indicating that product complexity or logistics might influence delivery times.

* Some vendors like CIPLA LIMITED or SCMS from RDC show less negative delays, implying they are closer to the planned dates.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Business Impact:**

* Helps identify reliable vendors for each product category.

* Allows targeted discussions with vendors having consistent delays.

* Can support contract negotiation based on performance.

* Aids in forecasting risks tied to specific vendors or products.

**Negative Business Impact:**

* Persistent delays by certain vendors may affect drug availability, especially for life-saving medicines.

* Could lead to stockouts in healthcare facilities and damage the reputation of supply chain management.

* Delays in high-priority products like ACT (for malaria) could lead to serious public health consequences.



#### Chart - 10 On-Time Delivery Rate by Managed By Team (Bar Chart)

In [ ]:
# Chart - 10 visualization code

# Create an 'On Time' column: 1 if Delivery Delay (Days) == 0, else 0
df1['On Time'] = df1['Delivery Delay (Days)'].apply(lambda x: 1 if x == 0 else 0)

# Group by 'Managed By' and calculate on-time delivery rate
managed_by_performance = df1.groupby('Managed By')['On Time'].mean().reset_index()

# Plot using Plotly
fig = px.bar(managed_by_performance,
             x='Managed By',
             y='On Time',
             title='On-Time Delivery Rate by Managed By Team',
             text='On Time',
             labels={'On Time': 'On-Time Delivery Rate'},
             color='On Time',
             color_continuous_scale='Viridis')

# Show the figure
fig.show()

##### 1. Why did you pick the specific chart?

A bar chart (with color shading for on-time delivery rate) was selected because it clearly and effectively compares the on-time delivery performance of shipments managed by different teams. It makes it easy to visually identify variations in performance between teams, and the numerical annotations help stakeholders instantly grasp the exact delivery rates without needing to refer to the dataset. It’s a simple, intuitive, and business-friendly visualization for categorical comparisons.

##### 2. What is/are the insight(s) found from the chart?

*	The South Africa Field Office achieved a significantly higher on-time delivery rate (98.25%) compared to PMO - US (55.67%).
*	There is a clear performance gap in delivery timeliness based on which team manages the shipment.
*	This suggests that operational practices or local vendor coordination in the South Africa Field Office are notably more effective in ensuring on-time deliveries.

3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Business Impact:**

*	Identifying that the South Africa Field Office has a much better delivery performance provides an opportunity to benchmark their processes and possibly standardize their effective practices across other teams, like PMO - US.
*	By adopting similar coordination, vendor management, or logistics strategies, the company can potentially improve overall delivery performance, leading to:
*	Increased client satisfaction
*	Reduced penalties for late deliveries
*	Better brand reputation in international logistics

**Negative Business Impact:**

*	The underperformance of PMO - US highlights a risk area where over 44% of shipments are delayed.
39*-+
This could lead to:
*	Client dissatisfaction or churn
*	Potential financial penalties for late deliveries
*	Increased operational costs due to re-shipping or expedited services
*	Negative impact on future contracts or service level agreements (SLAs)

Chart - 11 On-Time Delivery Rate by Shipment Mode (Bar Chart)

In [ ]:
# Chart - 11 visualization code

# Create an 'On Time' flag
df1['On Time'] = df1['Scheduled Delivery Date'] == df1['Delivered to Client Date']

# Group by Shipment Mode and calculate on-time delivery rate
shipment_performance = df1.groupby('Shipment Mode')['On Time'].mean().reset_index()

# Plotly bar chart
fig = px.bar(
    shipment_performance,
    x='Shipment Mode',
    y='On Time',
    text='On Time',
    title='On-Time Delivery Rate by Shipment Mode',
    labels={'On Time': 'On-Time Delivery Rate'},
    color='On Time'
)

fig.show()

##### 1. Why did you pick the specific chart?

A bar chart is an effective choice for comparing the on-time delivery rates across different shipment modes (Air, Air Charter, Ocean, and Truck) because it allows for clear visual comparison of discrete categories. The height of each bar directly corresponds to the on-time delivery rate, making it easy to identify which shipment mode performs best or worst. The use of different colors also helps distinguish between the categories, enhancing readability.

##### 2. What is/are the insight(s) found from the chart?

* Ocean shipment has the highest on-time delivery rate (78%), indicating it is the most reliable mode among those listed.
*  Air Charter has the lowest on-time delivery rate (19%), suggesting significant delays or inefficiencies in this mode.
*  Air (67%) and Truck (47%) fall in the middle, with Air performing better than Truck but still below Ocean.
*  There is a wide variation in performance, with Ocean outperforming Air Charter by nearly 60 percentage points, highlighting potential inconsistencies in operational efficiency across shipment modes.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Impact on Business:**

*  High Reliability of Ocean Shipments: The strong performance of Ocean shipments (78%) can be leveraged to build customer trust and satisfaction, potentially increasing repeat business and positive word-of-mouth.
*  Opportunity for Optimization: The significant gap between the best (Ocean) and worst (Air Charter) modes suggests room for improvement. Focusing on enhancing Air Charter efficiency could lead to cost savings and better service levels.
*  Data-Driven Decisions: The chart provides clear evidence to prioritize resources toward the most reliable modes (e.g., Ocean) and investigate underperforming ones (e.g., Air Charter).

**Negative Impact on Business:**
*  Poor Performance of Air Charter: The low on-time delivery rate (19%) could lead to customer dissatisfaction, lost contracts, or penalties, especially if this mode is critical for time-sensitive deliveries.
*  Inconsistent Service: The wide range of performance across modes (19% to 78%) may confuse customers or damage the company’s reputation for reliability if not addressed.
*  Resource Misallocation: Continued reliance on underperforming modes like Air Charter without improvement could waste resources and increase operational costs.

#### Chart - 12 Average Delivery Time by Country (Bar Chart)

In [ ]:
# Chart - 12 visualization code

# Calculate the Delivery Time (Days)
df1['Scheduled Delivery Date'] = pd.to_datetime(df1['Scheduled Delivery Date'], errors='coerce')
df1['Delivery Recorded Date'] = pd.to_datetime(df1['Delivery Recorded Date'], errors='coerce')
df1['Delivery Time (Days)'] = (df1['Delivery Recorded Date'] - df1['Scheduled Delivery Date']).dt.days


# Group by country and calculate the average delivery time
avg_delivery_by_country = df1.groupby('Country')['Delivery Time (Days)'].mean().reset_index()

# Create the Plotly bar chart
fig = px.bar(
    avg_delivery_by_country,
    x='Country',
    y='Delivery Time (Days)',
    title='Average Delivery Time by Country',
    labels={'Delivery Time (Days)': 'Average Delivery Time (Days)', 'Country': 'Country'},
)

fig.update_layout(xaxis_title='Country', yaxis_title='Average Delivery Time (Days)')
fig.show()

##### 1. Why did you pick the specific chart?

A bar chart was chosen to clearly compare average delivery times across countries, with bars showing positive (delays), negative (early), or zero (on-time) values, allowing quick identification of performance trends.

##### 2. What is/are the insight(s) found from the chart?

* Two countries show significant delays (around 20 and 15 days).
*  Most countries have early deliveries (negative values, e.g., -10 to -5 days).
*  No country appears on time (0), indicating inconsistent delivery performance.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Impact:**

 Early deliveries (negative values) in most countries can enhance customer satisfaction and build trust.

**Negative Impact:**

 Large delays (e.g., 20 and 15 days) in some countries risk customer loss and reputational damage.

#### Chart - 13 Delivery Performance vs Time Between PO and Scheduled Delivery (Scatter Plot)

In [ ]:
# Chart - 13 visualization code

# Calculate the time difference between PO sent date and scheduled delivery date
df1['Time_to_Delivery'] = (df1['Scheduled Delivery Date'] - df1['PO Sent to Vendor Date']).dt.days

# Calculate delivery performance (e.g., on-time or late)
df1['Delivery_Performance'] = (df1['Delivery Recorded Date'] <= df1['Scheduled Delivery Date']).astype(int)


# Create the scatter plot
fig = px.scatter(df1,
                 x='Time_to_Delivery',
                 y='Delivery_Performance',
                 title='Delivery Performance vs Time Between PO and Scheduled Delivery',
                 labels={'Time_to_Delivery': 'Time to Delivery (Days)',
                         'Delivery_Performance': 'Delivery Performance (0=Late, 1=On-Time)'},
                 trendline="ols") # Add a trendline to see the relationship

fig.show()


##### 1. Why did you pick the specific chart?

A scatter plot with a trendline was chosen to explore the relationship between the time it takes from when a purchase order (PO) is sent to a vendor and the scheduled delivery date, and the ultimate delivery performance (on-time or late).  This visualization allows us to identify if there's a correlation between the lead time and the likelihood of on-time delivery.  The trendline provides a visual representation of this correlation, making it easier to spot potential patterns or outliers.

##### 2. What is/are the insight(s) found from the chart?

*  Delivery performance starts near 1 (100% on-time) when time to delivery is around -200 days and decreases as time increases.
*  Performance drops steadily, reaching around 0.4 (40% on-time) at 600 days.
*  Most data points cluster around 0 to 300 days, with a noticeable decline in performance as time extends.

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

**Positive Impact:**

 Early scheduling (negative days) shows high on-time rates, suggesting benefits from proactive planning; identifies a threshold (around 300 days) for maintaining good performance.

**Negative Impact:**

 Longer lead times (e.g., 600 days) significantly reduce on-time delivery, risking customer dissatisfaction and potential penalties.

#### Chart - 14 Delivery Delay by Vendor INCO Term (Box Plot)



In [ ]:

# Convert date columns to datetime
df1["Scheduled Delivery Date"] = pd.to_datetime(df["Scheduled Delivery Date"], errors='coerce')
df1["Delivered to Client Date"] = pd.to_datetime(df["Delivered to Client Date"], errors='coerce')

# Calculate delivery delay
df1["Delivery Delay (Days)"] = (df1["Delivered to Client Date"] - df1["Scheduled Delivery Date"]).dt.days

# Drop rows with missing values in relevant columns
df1_box = df1.dropna(subset=["Vendor INCO Term", "Delivery Delay (Days)"])

# Create box plot
fig = px.box(
    df1_box,
    x="Vendor INCO Term",
    y="Delivery Delay (Days)",
    color="Vendor INCO Term",
    title="Delivery Delay by Vendor INCO Term",
    labels={"Delivery Delay (Days)": "Delivery Delay (Days)", "Vendor INCO Term": "INCO Term"}
)

fig.update_layout(
    xaxis_title="Vendor INCO Term",
    yaxis_title="Delivery Delay (Days)",
    showlegend=False
)

fig.show()


##### 1. Why did you pick the specific chart?

I chose the boxplot because it effectively visualizes the distribution and variability of delivery delays across different Vendor INCO Terms. This type of plot is ideal for identifying outliers, medians, and interquartile ranges, allowing a clear comparison of vendor delivery performance based on INCO terms.


##### 2. What is/are the insight(s) found from the chart?

* Some INCO terms like DDP and N/A - From RDC show greater variability and a higher concentration of extreme delays, indicating potential inefficiencies or logistical challenges.

* INCO terms like DAP and DDU show minimal variation in delivery delays, suggesting more consistent performance.

* This suggests that the type of INCO Term used by vendors can have a significant impact on delivery performance, and optimizing or standardizing INCO terms may lead to better control over delivery timelines.

**Positive Impact**

* DAP, DDU, and CIP terms show relatively low variability and delivery delays centered around zero or mildly negative values.

* These terms likely indicate well-managed logistics, with the buyer/vendor responsibilities clearly defined, resulting in more reliable and consistent delivery times.

**Negative Impact:**

* DDP and N/A - From RDC exhibit high variability and several extreme negative outliers (e.g., -300+ days).

* This suggests a lack of control or unpredictability in shipments, possibly due to complex customs processes, vendor delays, or unclear responsibility transitions.

* In particular, the “N/A - From RDC” category might reflect internal supply chain delays or incomplete documentation, making it harder to track performance.

#### Chart - 15 Shipment Weight vs Line Item Insurance Cost (Scatter Plot)

In [ ]:
# Visualization code

fig = px.scatter(
    df1,
    x='Weight (Kilograms)',  # Changed from 'Shipment Weight (Kilograms)'
    y='Line Item Insurance (USD)',
    trendline='ols',  # Ordinary Least Squares regression line
    title='Shipment Weight vs Line Item Insurance Cost',
    labels={
        'Weight (Kilograms)': 'Shipment Weight (kg)',  # Adjusted label accordingly
        'Line Item Insurance (USD)': 'Insurance Cost (USD)'
    },
    color_discrete_sequence=['indigo']
)
fig.update_layout(template='plotly_white')
fig.show()

##### 1. Why did you pick the specific chart?

I chose a scatter plot with a trendline because it effectively visualizes the relationship between two continuous numerical variables—shipment weight (kg) and insurance cost (USD). The scatter plot helps identify whether there's a pattern or correlation between the weight of the shipment and the insurance cost, while the trendline gives a clear direction of the relationship (positive, negative, or none).

##### 2. What is/are the insight(s) found from the chart?

* There is a positive correlation between shipment weight and insurance cost.

* As the weight increases, the insurance cost tends to increase as well, although there is some variability.

* The trendline indicates that, on average, heavier shipments incur higher insurance costs.

* However, there's a wide spread of insurance values at lower weights, suggesting other factors may also influence the cost.



**Positive Impact:**

The chart shows that weight is positively correlated with insurance cost, which aligns with expectations—heavier shipments generally carry more value or risk, justifying higher insurance.

**Negative Impact:**

 If not managed well, increasing shipment weights could significantly raise insurance expenses, impacting overall shipment cost performance.

## **5. Solution to Business Objective**

#### What do you suggest the client to achieve Business Objective ?
Explain Briefly.

**Control Insurance Costs :-**
Analyze weight vs. insurance trends to apply optimized pricing and reduce outliers.

**Reduce Shipment Delays :-**
Identify delay patterns by region/vendor and improve scheduling accuracy.

**Improve Vendor Performance :-**
Score vendors based on delivery and cost efficiency; switch or renegotiate with poor performers.

**Cut Freight & Operational Costs :-**
Recommend cost-effective shipping methods and consolidate low-weight shipments.

**Boost Customer Satisfaction :-**
Improve delivery accuracy and provide real-time tracking and updates.

**Forecast Demand Better :-**
Use historical trends to predict shipment volumes and prepare for peak periods.

**Ensure INCO Terms Compliance :-**
Standardize and monitor terms to avoid errors and shipment issues.

# **Conclusion**

Based on the analysis of delivery performance across various factors, several key insights have emerged that can positively impact the client's business.  Optimizing shipment modes, particularly by addressing the significant underperformance of Air Charter, and standardizing vendor INCO terms are crucial steps.  Furthermore, leveraging the strong performance of Ocean shipments and the positive correlation between weight and insurance costs will allow for better resource allocation and cost control.  Addressing regional disparities in delivery times and improving the relationship with underperforming vendors are also critical to improving overall performance.  By proactively implementing these solutions, the client can achieve better delivery performance, reduce costs, and enhance customer satisfaction.


### ***Hurrah! You have successfully completed your EDA Capstone Project !!!***